### PR TITLE
Scalar collections fixes

### PR DESCRIPTION
**Pull request by teclator**
- It adds an specific widget for dictionaries with primitive values like:

Formula:

```yaml
user_passwords:
    $type: edit-group
    $minItems: 1
    $prototype:
        $key:
            $type: text
        $type: text
    $default:
        alice: secret-password
        bob: you-shall-not-pass
```
Pillar:

```yaml
user_passwords:
    alice:
        secret-password
    bob:
        you-shall-not-pass
```

By now we are only supporting "text" ones

**Pending:** Reading and writing the pillar data

### CODE REVIEW

File: src/lib/y2configuration_management/salt/form.rb [hunk 1]
```diff
@@ -129,6 +130,18 @@ def locator
       def humanize(s)
         s.split(/[-_]/).reject(&:empty?).map(&:capitalize).join(" ")
       end
+
+      # Returns the type for a given form element specification
+      #
+      # @param spec [Hash] Form element specification
+      # @return [Symbol] Form element type
+      def type_for(_id, spec)
```
Comment (imobachgs) [inline src/lib/y2configuration_management/salt/form.rb]: Why are you passing the `_id`?

Comment (teclator) [reply src/lib/y2configuration_management/salt/form.rb]: leftover ;)

File: src/lib/y2configuration_management/widgets/key_value.rb [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+        true
+      end
+
+    private
+
+      # Input field for the hash $key
+      class Key < ::CWM::InputField
+        def label
+          "$key"
```
Comment (mvidner) [inline src/lib/y2configuration_management/widgets/key_value.rb]: Huh, this is not a nice user-visible label.

The form can provide the label, like here https://gitlab.suse.de/SLEPOS/SUMA_Retail/blob/d5709564b5c19da9b00606192789faf722922177/bind-formula/form.yml#L13
```yml
      $prototype:
        $type: text
        $key:
          $type: text
          $name: Option
```

File: src/lib/y2configuration_management/salt/form_data_reader.rb [hunk 3]
```diff
@@ -29,7 +29,37 @@ module Salt
...
+    #   [{ "$key" => "yast2", "url" => "https://yast.opensuse.org", "license" => "GPL" }]
+    #
+    # Something similar applies to hash based simple collections which are originally like this (it
+    # is a hash where the keys are specified by the user):
+    #
+    #   { "vers" => "4", "timeout" => "0" }
+    #
+    # It will be converted to:
+    #
+    #   [{ "$key" => "vers", "$value" => "4" }, { "$key" => "timeout", "$value" => "0" }]
```
[the remark this answers is not in the document]
Comment (imobachgs) [reply src/lib/y2configuration_management/salt/form_data_reader.rb]: Yes, you are right. I will try that approach.

**Pull request merged**